datalad/support/gitrepo.py

```diff
@@ -2907,7 +2908,7 @@ def get_content_info(self, paths=None, ref=None, untracked='all',
                 expect_fail=True)
         except CommandError as exc:
             if "fatal: Not a valid object name" in str(exc):
```

Comment:
Not introduced by this PR, but this could easily fail on py2:

```
$ datalad diff -f HEAD~ -t β-doesnt-exist
[ERROR  ] 'ascii' codec can't encode character u'\u03b2' in position 387: ordinal not in range(128) [gitrepo.py:get_content_info:2910] (UnicodeEncodeError)
```

If we were to fix that:

```diff
diff --git a/datalad/support/gitrepo.py b/datalad/support/gitrepo.py
index e9df21970..ff9ba6477 100644
--- a/datalad/support/gitrepo.py
+++ b/datalad/support/gitrepo.py
@@ -2907,7 +2907,7 @@ def get_content_info(self, paths=None, ref=None, untracked='all',
                 # we don't want it to scream on stdout
                 expect_fail=True)
         except CommandError as exc:
-            if "fatal: Not a valid object name" in str(exc):
+            if "fatal: Not a valid object name" in text_type(exc):
                 raise InvalidGitReferenceError(ref)
             raise
         lgr.debug('Done query repo: %s', cmd)
```

then we'd get an error in code that this PR touched (though a similar error would be there before this PR):

```
$ datalad diff -f HEAD~ -t β-doesnt-exist
[ERROR  ] 'ascii' codec can't decode byte 0xce in position 15: ordinal not in range(128) [diff.py:_diff_ds:261] (UnicodeDecodeError) 
```

To complete the fix, we could (1) ensure to/fr are unicode when they first come in and (2) make `InvalidGitReferenceError.__str__` return unicode on py2 (ideally it'd use six.python_2_unicode_compatible or something like it, but we don't currently do that for any of our `__str__` methods).

Reply:
Nice find. So - should we deal with it here or after? Could you post suggested changes?